src/flow.c

```diff
@@ -344,6 +345,24 @@ static inline void FlowUpdateTtlTC(Flow *f, uint8_t ttl)
     f->max_ttl_toclient = MAX(f->max_ttl_toclient, ttl);
 }
 
+static inline void FlowUpdateFlowRate(
+        ThreadVars *tv, DecodeThreadVars *dtv, Flow *f, const Packet *p, int dir)
+{
+    if (FlowRateStorageEnabled()) {
+        FlowRateStore *frs = FlowGetStorageById(f, FlowRateGetStorageID());
```

Comment:
should we check `f->flags & FLOW_IS_ELEPHANT` before the call to FlowGetStorageById?

Reply:
Indeed that's much better